From 4220e0558beba2dbc3a159298db8c80ad7b2ce21 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Fri, 13 Aug 2021 11:36:58 +0300
Subject: [PATCH] [SW-36134] preboot: gaudi2: turn on yellow led

In case of Gaudi2 there are multiple boot stages and 2 LEDs only are
available to indicate boot status. It is aligned with system to turn on
yellow LED(GPIO1) at preboot stage and turn off at Linux stage

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I2cb64c3138589e7396d89597a3c1aedf3caa4d22
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 357d391..275a285 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -30,18 +30,17 @@
 	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
 }
 
-/* TODO:configure yellow LED(GPIO2) in output mode and turn on(active low) */
+#define YELLOW_LED_GPIO		1
+
+/* Turn on yellow led @ GPIO1 (active low) */
 void preboot_led_on(void)
 {
-	/* TODO: Check this */
-	u32 val = BIT(2);
+	u32 val = BIT(YELLOW_LED_GPIO);
 
-	/* configure GPIO in output mode */
 	regmap_writel(val, mmPSOC_GPIO0_OUTENSET);
 
-	/* Turn ON LED (pin is active low) */
 	val = regmap_readl(mmPSOC_GPIO0_DATAOUT);
-	val &= ~BIT(2);
+	val |= BIT(YELLOW_LED_GPIO);
 	regmap_writel(val, mmPSOC_GPIO0_DATAOUT);
 }
 
